contracts/contracts/buyback/Buyback.sol

```diff
@@ -1,74 +1,59 @@
 // SPDX-License-Identifier: agpl-3.0
 pragma solidity ^0.8.0;
 
-import { Governable } from "../governance/Governable.sol";
+import { Strategizable } from "../governance/Strategizable.sol";
 import "../interfaces/chainlink/AggregatorV3Interface.sol";
 import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 import { SafeMath } from "@openzeppelin/contracts/utils/math/SafeMath.sol";
 import { UniswapV3Router } from "../interfaces/UniswapV3Router.sol";
 
-contract Buyback is Governable {
+contract Buyback is Strategizable {
     using SafeERC20 for IERC20;
     using SafeMath for uint256;
 
     event UniswapUpdated(address _address);
-    event BuybackFailed(bytes data);
 
     // Address of Uniswap
     address public uniswapAddr;
 
-    // Address of OUSD Vault
-    address public immutable vaultAddr;
-
     // Swap from OUSD
     IERC20 immutable ousd;
 
-    // Swap to OGN
-    IERC20 immutable ogn;
+    // Swap to OGV
+    IERC20 immutable ogv;
 
     // USDT for Uniswap path
     IERC20 immutable usdt;
 
     // WETH for Uniswap path
     IERC20 immutable weth9;
 
-    // Oracles
-    address immutable ognEthOracle;
-    address immutable ethUsdOracle;
+    // Address that receives rewards
+    address public immutable rewardsSource;
 
     constructor(
         address _uniswapAddr,
-        address _vaultAddr,
+        address _strategistAddr,
         address _ousd,
-        address _ogn,
+        address _ogv,
         address _usdt,
         address _weth9,
-        address _ognEthOracle,
-        address _ethUsdOracle
+        address _rewardsSource
     ) {
         uniswapAddr = _uniswapAddr;
-        vaultAddr = _vaultAddr;
+        strategistAddr = _strategistAddr;
         ousd = IERC20(_ousd);
-        ogn = IERC20(_ogn);
+        ogv = IERC20(_ogv);
         usdt = IERC20(_usdt);
         weth9 = IERC20(_weth9);
-        ognEthOracle = _ognEthOracle;
-        ethUsdOracle = _ethUsdOracle;
+        rewardsSource = _rewardsSource;
         // Give approval to Uniswap router for OUSD, this is handled
         // by setUniswapAddr in the production contract
         IERC20(_ousd).safeApprove(uniswapAddr, 0);
```

Comment:
Turns out we don't actually need this approve to zero here, since OUSD is happy approving to different amounts. 

Also, since this contract is not upgradable, nothing will have run here before setting an approval.

Reply:
@shahthepro can you address this comment?

Reply:
Yep, it isn't needed in the constructor. But might still need in the `setUniswapAddr()` function. So, I'm removing this from the constructor but leaving that method as is.